extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -323,16 +334,26 @@ export class ResourceTypePickerDialog extends DialogBase {
 
 	private async installTools(): Promise<void> {
 		this._installToolButton.enabled = false;
-		let i: number = 0;
+		let tool: ITool;
 		try {
-			for (; i < this._tools.length; i++) {
-				if (this._tools[i].needsInstallation) {
+			const toolRequirements = this.getCurrentProvider().requiredTools;
+			for (let i: number = 0; i < toolRequirements.length; i++) {
+				const toolReq = toolRequirements[i];
+				tool = this.toolsService.getToolByName(toolReq.name)!;
+				if (tool.needsInstallation) {
 					// Update the informational message
 					this._dialogObject.message = {
 						level: azdata.window.MessageLevel.Information,
 						text: localize('deploymentDialog.InstallingTool', "Required tool '{0}' [ {1} ] is being installed now.", this._tools[i].displayName, this._tools[i].homePage)
 					};
 					await this._tools[i].install();
+					if (tool.isInstalled && toolReq.version && !tool.isSameOrNewerThan(toolReq.version)) {
```

Comment:
hmm, this shouldn't happen, should ony be a dev time scenario, you updated the required version but not the install tool method, you can treat it as a dev time error.

Reply:
yeah perhaps the check ensures that we still do have the correct version before proceeding so it is a good defensive code.